thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/EventDTO.java

```diff
@@ -1,7 +1,24 @@
 package com.linkedin.thirdeye.datalayer.dto;
 
 import com.linkedin.thirdeye.datalayer.pojo.EventBean;
+import java.util.Objects;
+
 
 public class EventDTO extends EventBean {
 
+  @Override
+  public int hashCode() {
+    return Objects.hash(getName(), getStartTime(), getEndTime(), getEventType());
```

Comment:
I strongly suggest to include id in equals() and hashCode(). Otherwise, we might see some strage side-effects with the storage backend